src/Kernel32.Tests/Kernel32.cs

```diff
@@ -149,4 +152,425 @@ public void QueryFullProcessImageName_CanGetForCurrentProcess()
+        try
+        {
+            const int testDataSize = 256;
+            var buffer = new byte[testDataSize];
+
+            using (var file = CreateFile(
+                testPath,
+                PInvoke.Kernel32.FileAccess.GenericWrite,
+                PInvoke.Kernel32.FileShare.None,
+                IntPtr.Zero,
+                CreationDisposition.OpenExisting,
+                CreateFileFlags.OverlappedFlag,
+                new SafeObjectHandle()))
+            {
+                var overlapped = default(OVERLAPPED);
+                fixed (byte* pExpected = buffer)
+                {
+                    var result = WriteFile(file, pExpected, testDataSize, null, &overlapped);
+                    if (result)
+                    {
+                        // We can't really test anything not covered by another test here :(
+                        return;
+                    }
+
+                    var lastError = GetLastError();
+                    Assert.Equal(Win32ErrorCode.ERROR_IO_PENDING, lastError);
+                    try
+                    {
+                        var cancelled = CancelIoEx(file, null);
+
+                        // We can't assert that it's true as if the IO finished already it'll fail with ERROR_NOT_FOUND
+                        if (!cancelled)
+                        {
+                            Assert.Equal(Win32ErrorCode.ERROR_NOT_FOUND, GetLastError());
+                        }
+                    }
+                    finally
+                    {
+                        uint bytesTransfered;
+                        GetOverlappedResult(file, &overlapped, out bytesTransfered, true);
+                    }
+                }
+            }
+        }
+        finally
+        {
+            File.Delete(testPath);
+        }
+    }
+
+    [Fact]
+    public unsafe void CancelIoEx_CancelWriteSpecific()
+    {
+        var testPath = Path.GetTempFileName();
+        try
+        {
+            const int testDataSize = 256;
+            var buffer = new byte[testDataSize];
+
+            using (var file = CreateFile(
+                testPath,
+                PInvoke.Kernel32.FileAccess.GenericWrite,
+                PInvoke.Kernel32.FileShare.None,
+                IntPtr.Zero,
+                CreationDisposition.OpenExisting,
+                CreateFileFlags.OverlappedFlag,
+                new SafeObjectHandle()))
+            {
+                var overlapped = default(OVERLAPPED);
+                fixed (byte* pExpected = buffer)
+                {
+                    var result = WriteFile(file, pExpected, testDataSize, null, &overlapped);
+                    if (result)
+                    {
+                        // We can't really test anything not covered by another test here :(
+                        return;
+                    }
+
+                    var lastError = GetLastError();
+                    Assert.Equal(Win32ErrorCode.ERROR_IO_PENDING, lastError);
```

Comment:
The test can't fail but it can test nothing really in one case :
* If IO finishes synchronously, then `WriteFile` return true and the rest finishes, never even calling `CancelIoEx`

Also it can test 2 different things after that (both useful but having the ability to drive and have 2 different tests would be beter, that's #41)
* If `CancelIoEx` is called before the IO finishing it'll test cancellation
* Otherwise it'll test that we get the correct return value (`ERROR_NOT_FOUND`) indicating that the IO already finished by the time `CancelIoEx` was called